controllers/vmset_controller.go

```diff
@@ -109,6 +109,52 @@ func (r *VMSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
+	// What VMs do we currently have for this VMSet?
+	virtualMachineList, err := common.GetAllVirtualMachinesWithLabel(r, map[string]string{
+		OwnerNameLabelSelector: instance.Name,
```

Comment:
not required for this change, but we probably want to add a role label to the VMs and filter here on the role specific VMs which is needed when we start to support split composable roles into different vmsets

Reply:
Would the uniqueness of `instance.Name` + `instance.Namespace` not be enough to ensure the VMs are filtered properly?  Or are you saying elsewhere where we might be filtering VMs and might not have the instance name/namespace available?

Reply:
yes it should, you are right.